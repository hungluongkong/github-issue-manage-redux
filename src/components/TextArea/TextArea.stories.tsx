import React from 'react';
import { storiesOf } from '@storybook/react';
import TextArea from '.';
import { PreviewArea, Code, DetailTable } from '../commonStyled';

// Detail of props
const propsDetail: JSX.Element = DetailTable({
  data: [
    { name: 'defaultValue', type: 'String' },
    { name: 'placeholder', type: 'String' },
    { name: 'isRequire', type: 'boolean', default: 'false' },
    { name: 'textRef', type: 'React.RefObject<HTMLTextAreaElement>' },
  ],
});

storiesOf('TextArea', module).add('Default', () => (
  <>
    <h3>Default TextArea</h3>
    <PreviewArea>
      <TextArea defaultValue="default text" placeholder="Your input here" />
    </PreviewArea>
    <Code>
      {`
          <TextArea
            defaultValue="default text"
            placeholder="Type your input here"
            textRef={ref}
          />
        `}
    </Code>
    {propsDetail}
  </>
));
