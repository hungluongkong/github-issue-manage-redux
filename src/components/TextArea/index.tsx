import React from 'react';
import styled from 'styled-components';
import { InputProps } from '../../types';
import * as color from '../../theme/color';

const StyledTextArea = styled.textarea`
  width: 100%;
  height: 200px;
  padding: 5px 12px;
  resize: vertical;
  flex-grow: 1;
  background-color: ${color.INPUT_BG_COLOR};
  line-height: 20px;
  border: 1px solid ${color.INPUT_BORDER_COLOR};
  border-radius: 6px;
  box-sizing: border-box;

  :focus {
    box-shadow: 1px 1px 1px 1px ${color.SHADOW_BORDER};
  }
`;

const TextArea = ({
  textRef,
  defaultValue = '',
  isRequire = false,
  onChange = () => {},
}: InputProps): JSX.Element => (
  <StyledTextArea
    ref={textRef as React.RefObject<HTMLTextAreaElement>}
    onChange={(e: React.ChangeEvent<HTMLTextAreaElement>) => onChange(e.target.value)}
    defaultValue={defaultValue}
    required={isRequire}
  />
);

TextArea.defaultProps = {
  isRequire: false,
};

export default TextArea;
