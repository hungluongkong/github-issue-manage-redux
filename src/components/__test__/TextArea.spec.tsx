import React from 'react';
import { create } from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import TextArea from '../TextArea';

describe('Unit test for TextArea component', () => {
  it('snapshot', () => {
    const textArea = create(<TextArea />);
    expect(textArea).toMatchSnapshot();
  });

  it('render Default value', () => {
    const defaultValue = 'Test text';
    const textArea = shallow(<TextArea defaultValue={defaultValue} />);
    expect(textArea.prop('defaultValue')).toEqual(defaultValue);
  });

  it('simulate on change', () => {
    const defaultValue = 'Default text';
    const newValue = 'Changed text';
    const onChange = jest.fn();

    const input = mount(<TextArea defaultValue={defaultValue} onChange={onChange} />);

    input.simulate('change', { target: { value: newValue } });

    expect(onChange).toBeCalledWith(newValue);
  });
});
