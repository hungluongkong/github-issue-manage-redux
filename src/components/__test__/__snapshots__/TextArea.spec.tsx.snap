// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for TextArea component snapshot 1`] = `
<textarea
  className="sc-AxjAm hMroFO"
  defaultValue=""
  onChange={[Function]}
  required={false}
/>
`;
